scout/adapter/mongo/panel.py

```diff
@@ -549,23 +549,36 @@ def latest_panels(self, institute_id, include_hidden=False):
     def clinical_symbols(self, case_obj):
         """Return all the clinical gene symbols for a case."""
         panel_ids = [panel["panel_id"] for panel in case_obj.get("panels", [])]
-        query = self.panel_collection.aggregate(
+        query_result = self.panel_collection.aggregate(
             [
                 {"$match": {"_id": {"$in": panel_ids}}},
                 {"$unwind": "$genes"},
                 {"$group": {"_id": "$genes.symbol"}},
             ]
         )
-        return set(item["_id"] for item in query)
+        return set(item["_id"] for item in query_result)
 
     def clinical_hgnc_ids(self, case_obj):
         """Return all the clinical gene hgnc IDs for a case."""
         panel_ids = [panel["panel_id"] for panel in case_obj.get("panels", [])]
-        query = self.panel_collection.aggregate(
+        query_result = self.panel_collection.aggregate(
             [
                 {"$match": {"_id": {"$in": panel_ids}}},
                 {"$unwind": "$genes"},
                 {"$group": {"_id": "$genes.hgnc_id"}},
             ]
         )
-        return set(item["_id"] for item in query)
+        return set(item["_id"] for item in query_result)
+
+    def search_panels(self, search_string):
+        """Return all panels and versions that contain given gene, list is sorted"""
+
+        # Try to cast search string to integer for searching hgnc_id:s.
+        try:
+            search_int = int(search_string)
+        except ValueError as _err:
+            search_int = search_string
+        query = {"$or": [{"genes.hgnc_id": search_int}, {"genes.symbol": search_string}]}
+        result = self.panel_collection.find(query)
+        result_list = [[element["panel_name"], element["version"]] for element in result]
```

Comment:
What you'll get after this query is this: ['name', 1.0, 'name', 2.0]

But what you need to display results is this: [['name', [1.0, 2.0]]]

And to reformat results you have created 2 new functions (_compile_panel_versions and _compile_panel_versions) below. 

You don't need all that code if you change your query to return just what you need instead. Try using an aggregation:

```
[{
    $match: {
        'genes.hgnc_id': <HGNC-iD>
    }
}, {
    $group: {
        _id: {
            display_name: '$display_name'
        },
        versions: {
            $addToSet: '$version'
        }
    }
}, {
    $sort: {
        display_name: 1
    }
}]
```

That will return results like this:

<img width="727" alt="image" src="https://user-images.githubusercontent.com/28093618/185884425-19bf1941-eda7-46f2-b5ef-0509b9ab293a.png">